qualcommax: ipq60xx: add support for Netgear WAX610 and WAX610Y

ServError opened the pull request:
Support depends on https://github.com/openwrt/firmware_qca-wireless/pull/73

Did not have a real WAX610Y on hand to test. Those builds were tested on the WAX610.

Netgear WAX610 / WAX610Y is a 802.11ax AP claiming AX1800 support.

Specifications:
---------------
* CPU: Qualcomm IPQ6018 Quad core Cortex-A53
* RAM: 512MB
* Storage: Winbond W29N01HZBINF 128MB NAND
* Ethernet:
  * 2.5GbE RJ45 port with PoE input
* WLAN:
  * 2.4GHz/5GHz
* LEDs:
  * Power: Dual-color LED (Green / Orange), Single-color LED (Blue)
  * Ethernet: Dual-color LED (Green / Orange)
  * 2.4Ghz WiFi: 2x Single-color LED (Blue, Green)
  * 5Ghz WiFi: 2x Single-color LED (Blue, Green)
* Buttons:
  * 1x Reset
* UART: 4-pin populated header
  * Pinout 1 - VCC, 2 - GND, 3 - TX, 4 - RX

Installation:
=============

Web UI method
-------------

Set up the device using the stock web UI in local management mode.

Download the ui-factory.tar image.
 * Enter the Management tab
 * Select Maintenance->Upgrade->Firmware Upgrade
 * Ensure the "Upgrade Options" field has "Local" selected
 * Click "Browse File" and select the image
 * Click "Upgrade" (NOT "Upgrade Now") at the bottom of the page

The device will install the image but reboot back into the vendor firmware. Return the to Firmware Upgrade page as above and select "Boot up Backup Firmware". The device should reboot into OpenWRT.

TFTP method
-----------

To flash via tftp, first place the initramfs image on the TFTP server.

    setenv serverip <ip of tftp server>
    setenv ipaddr <ip in same subnet as tftp server>
    tftpboot netgear_wax610-initramfs-uImage.itb
    bootm

This should boot OpenWRT. Once booted, flash the sysupgrade.bin image using either luci or the command line.

robimarko commented:
What is the difference between WAX610 and the Y version?

ServError commented:
> What is the difference between WAX610 and the Y version?

Y is an outdoor model with different antenna arrangement and different board file.

ServError commented:
> Please deduplicate the DTS, make a common DTSI and just add model specifics to their individual DTS-s

Done. ipq-wifi package makefile updated.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-wax610-base.dtsi:
```diff
@@ -0,0 +1,217 @@
+
+	switch_lan_bmp = <ESS_PORT5>;
+	switch_wan_bmp = <0>;
+	switch_mac_mode = <MAC_MODE_PSGMII>;
```
You dont need this since there is no QCA807x PHY being used

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-wax610-base.dtsi:
```diff
@@ -0,0 +1,217 @@
+	status = "okay";
+
+	switch_lan_bmp = <ESS_PORT5>;
+	switch_wan_bmp = <0>;
```
I dont think switch_wan_bmp is needed

ServError replied:
Without it I had been getting

`ssdk_dt_parse_port_bmp[1063]:INFO:port_bmp doesn't exist!`

and I wasn't sure if the lan mapping was being impacted, though it didn't seem so.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-wax610-base.dtsi:
```diff
@@ -0,0 +1,217 @@
+	switch_lan_bmp = <ESS_PORT5>;
+	switch_wan_bmp = <0>;
+	switch_mac_mode = <MAC_MODE_PSGMII>;
+	switch_mac_mode1 = <MAC_MODE_SGMII_CHANNEL0>;
```
Since this is connected to QCA8081 you want `MAC_MODE_SGMII_PLUS`

ServError replied:
Stock DTS called for

`switch_mac_mode = <0x00000000>;
switch_mac_mode1 = <0x0000000f>;`

Which mirrors the above, though of course vendor firmware could be changing it during runtime. Any idea why the discrepancy?

robimarko replied:
Stock DTS-es are full of invalid stuff, PSGMII on UNIPHY0 is QCA default for most boards so they just modify existing boards to "work" and call it a day.

UNIPHY0 is not being used as you dont have a QCA807x PHY on the bus, only QCA8081 which is connected to UNIPHY1 and since its 2.5G its SGMII+, but rarely do vendors actually set it like that due to outdated QSDK versions.

Trust me, you just need `MAC_MODE_SGMII_PLUS` for `switch_mac_mode1`

ServError replied:
Understood. The change is already in place, just wanted the background info if it was available, thanks.

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-wax610-base.dtsi:
```diff
@@ -0,0 +1,217 @@
+		compatible = "ethernet-phy-id004d.d101";
+		reg = <24>;
+		reset-deassert-us = <10000>;
+		reset-gpios = <&tlmm 77 GPIO_ACTIVE_LOW>;
```
You can move the reset GPIO to the bus level, then you dont need to define the PHY compatible manually as it can be autoprobed

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-wax610-base.dtsi:
```diff
@@ -0,0 +1,217 @@
+
+/ {
+	reserved-memory {
+		m3_dump_region: memory@50100000 {
```
Do you really need M3 dump?

AFAIK its not really even used in ath11k

ServError replied:
Likely not, I just carried it over from stock DTS. Removed.

ServError committed:
qualcommax: ipq60xx: add support for Netgear WAX610 and WAX610Y

Netgear WAX610 / WAX610Y is a 802.11ax AP claiming AX1800 support.

Specifications:
---------------
* CPU: Qualcomm IPQ6018 Quad core Cortex-A53
* RAM: 512MB
* Storage: Winbond W29N01HZBINF 128MB NAND
* Ethernet:
  * 2.5GbE RJ45 port with PoE input
* WLAN:
  * 2.4GHz/5GHz
* LEDs:
  * Power: Dual-color LED (Green / Orange), Single-color LED (Blue)
  * Ethernet: Dual-color LED (Green / Orange)
  * 2.4Ghz WiFi: 2x Single-color LED (Blue, Green)
  * 5Ghz WiFi: 2x Single-color LED (Blue, Green)
* Buttons:
  * 1x Reset
* UART: 4-pin populated header
  * Pinout 1 - VCC, 2 - GND, 3 - TX, 4 - RX

Installation:
=============

Web UI method
-------------

Set up the device using the stock web UI in local management mode.

Download the ui-factory.tar image.
 * Enter the Management tab
 * Select Maintenance->Upgrade->Firmware Upgrade
 * Ensure the "Upgrade Options" field has "Local" selected
 * Click "Browse File" and select the image
 * Click "Upgrade" (NOT "Upgrade Now") at the bottom of the page

The device will install the image but reboot back into the vendor
firmware. Return the to Firmware Upgrade page as above and select
"Boot up Backup Firmware". The device should reboot into OpenWRT.

TFTP method
-----------

To flash via tftp, first place the initramfs image on the TFTP server.

    setenv serverip <ip of tftp server>
    setenv ipaddr <ip in same subnet as tftp server>
    tftpboot netgear_wax610-initramfs-uImage.itb
    bootm

This should boot OpenWRT. Once booted, flash the sysupgrade.bin image
using either luci or the command line.

Signed-off-by: Erik Servili <serverror@serverror.com>
Link: https://github.com/openwrt/openwrt/pull/18377
Signed-off-by: Robert Marko <robimarko@gmail.com>

ServError commented:
Requested changes completed and tested.

robimarko commented:
Thanks! Rebased on top of main and merged!
